lightning/src/ln/channelmanager.rs

```diff
@@ -2853,55 +2903,7 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 												log_bytes!(payment_hash.0), payment_data.total_msat, inbound_payment.get().min_value_msat.unwrap());
 											fail_htlc!(claimable_htlc);
 										} else {
-											let mut total_value = 0;
-											let htlcs = channel_state.claimable_htlcs.entry(payment_hash)
-												.or_insert(Vec::new());
-											if htlcs.len() == 1 {
-												if let OnionPayload::Spontaneous(_) = htlcs[0].onion_payload {
-													log_trace!(self.logger, "Failing new HTLC with payment_hash {} as we already had an existing keysend HTLC with the same payment hash", log_bytes!(payment_hash.0));
-													fail_htlc!(claimable_htlc);
-													continue
-												}
-											}
-											htlcs.push(claimable_htlc);
-											for htlc in htlcs.iter() {
-												total_value += htlc.value;
-												match &htlc.onion_payload {
-													OnionPayload::Invoice(htlc_payment_data) => {
-														if htlc_payment_data.total_msat != payment_data.total_msat {
-															log_trace!(self.logger, "Failing HTLCs with payment_hash {} as the HTLCs had inconsistent total values (eg {} and {})",
-																				 log_bytes!(payment_hash.0), payment_data.total_msat, htlc_payment_data.total_msat);
-															total_value = msgs::MAX_VALUE_MSAT;
-														}
-														if total_value >= msgs::MAX_VALUE_MSAT { break; }
-													},
-													_ => unreachable!(),
-												}
-											}
-											if total_value >= msgs::MAX_VALUE_MSAT || total_value > payment_data.total_msat {
-												log_trace!(self.logger, "Failing HTLCs with payment_hash {} as the total value {} ran over expected value {} (or HTLCs were inconsistent)",
-													log_bytes!(payment_hash.0), total_value, payment_data.total_msat);
-												for htlc in htlcs.iter() {
-													fail_htlc!(htlc);
-												}
-											} else if total_value == payment_data.total_msat {
-												new_events.push(events::Event::PaymentReceived {
-													payment_hash,
-													purpose: events::PaymentPurpose::InvoicePayment {
-														payment_preimage: inbound_payment.get().payment_preimage,
-														payment_secret: payment_data.payment_secret,
-													},
-													amt: total_value,
-												});
-												// Only ever generate at most one PaymentReceived
```

Comment:
> Also, shouldn't we still remove the entry for users of the old API? Otherwise does the payment ever get removed?

Good catch. These payments would still have gotten removed on `best_block_updated` if they've expired, but that was a bug

Let me know if the docs updates are more what you're thinking.